src/test/components/Root.test.js

```diff
@@ -0,0 +1,126 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+
+import { ProfileViewWhenReady } from '../../components/app/Root';
+import { updateUrlState } from '../../actions/app';
+import {
+  fatalError,
+  temporaryError,
+  viewProfile,
+  waitingForProfileFromAddon,
+  waitingForProfileFromStore,
+} from '../../actions/receive-profile';
+import { stateFromLocation } from '../../app-logic/url-handling';
+import { TemporaryError } from '../../utils/errors';
+
+import { shallowWithStore } from '../fixtures/enzyme';
+import { blankStore } from '../fixtures/stores';
+import { getProfileFromTextSamples } from '../fixtures/profiles/make-profile';
+
+describe('app/ProfileViewWhenReady', function() {
+  it('renders an initial home', function() {
+    const { view } = setup();
+
+    // dive() will shallow-render the wrapped component
+    expect(view.dive()).toMatchSnapshot();
+  });
+
+  it('renders the addon loading page', function() {
+    const { view, dispatch, navigateToAddonLoadingPage } = setup();
+
+    navigateToAddonLoadingPage();
+    dispatch(waitingForProfileFromAddon());
+    expect(view.dive()).toMatchSnapshot();
+  });
+
+  it('renders the profile view', function() {
+    const { view, dispatch, navigateToStoreLoadingPage } = setup();
+
+    navigateToStoreLoadingPage();
+    dispatch(waitingForProfileFromStore());
+    expect(view.dive()).toMatchSnapshot();
+
+    const { profile } = getProfileFromTextSamples(`A`);
+    dispatch(viewProfile(profile));
+    expect(view.dive()).toMatchSnapshot();
+  });
+
+  it('renders temporary errors', function() {
+    const { view, dispatch, navigateToStoreLoadingPage } = setup();
+
+    navigateToStoreLoadingPage();
+    dispatch(
+      temporaryError(
+        new TemporaryError('This is a temporary error, wait a bit more.', {
+          count: 3,
+          total: 10,
+        })
+      )
+    );
+    expect(view.dive()).toMatchSnapshot();
+  });
+
+  it('renders an home when the user pressed back after an error', function() {
+    const {
+      view,
+      dispatch,
+      navigateToStoreLoadingPage,
+      navigateBackToHome,
+    } = setup();
+
+    navigateToStoreLoadingPage();
+    dispatch(fatalError(new Error('Error while loading profile')));
+    expect(view.dive()).toMatchSnapshot();
+    expect(console.error).toHaveBeenCalled();
+
+    navigateBackToHome();
+    expect(view.dive()).toMatchSnapshot();
+  });
+});
+
+function setup() {
+  // Let's silence the error output to the console
+  jest.spyOn(console, 'error').mockImplementation(() => {});
+
+  const store = blankStore();
+  const view = shallowWithStore(<ProfileViewWhenReady />, store);
+
+  function navigateToStoreLoadingPage() {
+    const newUrlState = stateFromLocation({
+      pathname: '/public/404ff08ee5f412f355264601f12e5feff0ebaee6/calltree',
```

Comment:
As I understand this, we are here testing for the route, but generating a profile from the fixtures. So we are not grabbing any of the actually profile data.

If we just need this for routing (so have a valid public url), I'd suggest to change it to something like `public/0000000000001/calltree` so it is obvious it is a mock. Otherwise test for a valid URL. 
The context here would be the error handling of the profile, so if we ever delete this existing profile, this could lead to confusions. Also it might be related to the errors mentioned above.

Reply:
done :)